src/test/java/com/artipie/rpm/asto/AstoRepoAddTest.java

```diff
@@ -129,9 +129,45 @@ void addsPackagesToRepo() throws IOException {
             this.storage.exists(new Key.From("lib", lib)).join(),
             new IsEqual<>(true)
         );
-        this.checkMeta("primary-res.xml", XmlPackage.PRIMARY);
-        this.checkMeta("other-res.xml", XmlPackage.OTHER);
-        this.checkMeta("filelists-res.xml", XmlPackage.FILELISTS);
+        MatcherAssert.assertThat(
+            "Primary xml should have three records",
+            new String(this.mbytes.value(XmlPackage.PRIMARY), StandardCharsets.UTF_8),
+            XhtmlMatchers.hasXPaths(
+                "/*[local-name()='metadata' and @packages='3']",
+                //@checkstyle LineLengthCheck (3 lines)
+                "/*[local-name()='metadata']/*[local-name()='package']/*[local-name()='name' and text()='abc']",
+                "/*[local-name()='metadata']/*[local-name()='package']/*[local-name()='name' and text()='time']",
+                "/*[local-name()='metadata']/*[local-name()='package']/*[local-name()='name' and text()='libnss-mymachines2']"
+            )
+        );
```

Comment:
I do not think this should be extracted to some method, thus we will add some logic and it would be more complicated to understand the check. I think that when it comes to expected part in the assertion, it should be as simple as possible, crystal clear and some code/stings duplication is much better than some extracted common logic.